### PR TITLE
Manual implementation of Clone

### PR DESCRIPTION
**Pull request by asonnino**
Manual implementation of Clone for all Proof structures to circumvent a known rust bug. See: https://github.com/rust-lang/rust/issues/41481

### CODE REVIEW

**Review by kevinlewi — COMMENTED** [review]
LGTM, but will leave it to @slawlor for the final approval in case this conflicts with anything

File: akd/src/proof_structs.rs [hunk 1]
```diff
@@ -33,9 +33,22 @@ pub struct MembershipProof<H: Hasher> {
     pub dirs: Vec<Direction>,
 }
 
+// Manual implementation of Clone, see: https://github.com/rust-lang/rust/issues/41481
+impl<H: Hasher> Clone for MembershipProof<H> {
+    fn clone(&self) -> Self {
+        Self {
+            label: self.label,
+            hash_val: self.hash_val,
```
Comment (slawlor) [inline akd/src/proof_structs.rs:41]: ```Hasher``` implements copy natively?

**Review by slawlor — APPROVED** [review]
Seems fine to me, just want to be sure about the types of ```Hasher```

File: akd/src/proof_structs.rs (hunk 1 above)
Comment (asonnino) [reply akd/src/proof_structs.rs:41]: Not `Hasher`, but `Hasher::Digest` does implement copy

**Pull request merged**